samples/client/petstore/r_test/R/PetApi.r

```diff
@@ -0,0 +1,161 @@
+# Swagger Petstore
+# 
+# This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.
+# 
+# OpenAPI spec version: 1.0.0
+# Contact: apiteam@swagger.io
+# Generated by: https://github.com/swagger-api/swagger-codegen.git
+
+PetApi <- R6::R6Class(
+  'PetApi',
+  public = list(
+    userAgent = "Swagger-Codegen/1.0.0/r",
+    host = NULL,
+    basePath = NULL,
+    scheme = NULL,
+    url = NULL,
+    initialize = function(host, basePath, scheme){
+      self$host <- host
+      self$basePath <- basePath
+      self$scheme <- scheme
+      self$url <- sprintf("%s://%s/%s/pet/", scheme, host, basePath)
+    },
+
+    add_pet = function(body){
+      resp <- httr::POST(paste0(self$url),
+          httr::add_headers("User-Agent" = self$userAgent, "accept" = "application/json", "content-type" = "application/xml")
+          ,body = body$toJSON()
+          )
+
+      if (httr::status_code(resp) >= 200 && httr::status_code(resp) <= 299) {
+        # void response, no need to return anything
+      } else if (httr::status_code(resp) >= 400 && httr::status_code(resp) <= 499){
+        Response$new("API client error", resp)
+      } else if (httr::status_code(resp) >= 500 && httr::status_code(resp) <= 599){
+        Response$new("API server error", resp)
+      }
+
+    },
+    delete_pet = function(pet_id, api_key){
+      resp <- httr::DELETE(paste0(self$url, pet_id),
+          httr::add_headers("User-Agent" = self$userAgent, "content-type" = "application/xml", "api_key" = api_key)
+          )
+
+      if (httr::status_code(resp) >= 200 && httr::status_code(resp) <= 299) {
+        # void response, no need to return anything
+      } else if (httr::status_code(resp) >= 400 && httr::status_code(resp) <= 499){
+        Response$new("API client error", resp)
+      } else if (httr::status_code(resp) >= 500 && httr::status_code(resp) <= 599){
+        Response$new("API server error", resp)
+      }
+
+    },
+    find_pets_by_status = function(status){
+      resp <- httr::GET(paste0(self$url),
+          httr::add_headers("User-Agent" = self$userAgent, "content-type" = "application/xml")
+          )
+      # TODO add support for query parameters "status"
```

Comment:
@ramnov I'm not sure how query string can be added to `httpr`? Is there something like `httr::add_queries`? I couldn't find one in https://www.rdocumentation.org/packages/httr/versions/1.2.1

Reply:
There is a query parameter in GET function. 
Example : GET("http://httpbin.org/cookies/set", query = list(a = 1, b = 2))
will translate to http://httpbin.org/cookies/set?a=1&b=1
More Examples here : https://stackoverflow.com/questions/32742535/difference-between-query-and-body-in-httrpost

Reply:
@ramnov OK. Will give that a try.